### ISSUE TITLE
telegraf-pi-get-throttled.sh throwing an error

### ISSUE THREAD

**Issue by radu-boboc** [fivesixzero/telegraf-pi-bash#5]
Hi,

I have 3x raspberries, and only one of them is under super high load, and reaching the max temp.
While doing that it throws this error: `printf: warning: 10000000000000000000: Numerical result out of range` and the script fails to do anything useful, like recording the data in influxdb

Running  `sudo /opt/vc/bin/vcgencmd get_throttled outputs this : throttled=0x80000` so I have a _Soft temperature limit has occurred_

I don't understand why printf throws that error and I don't know how to fix it.
```
vcgencmd version
Mar 24 2022 13:19:43
Copyright (c) 2012 Broadcom
version e5a963efa66a1974127860b42e913d2374139ff5 (clean) (release) (start_cd)

```

```
# printf %020d 10000000000000000000
-bash: printf: warning: 10000000000000000000: Numerical result out of range
092233720368547

#/usr/bin/printf --version
printf (GNU coreutils) 8.32
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by David MacKenzie.
```

Can you please help ?
Please let me know what additional data you need.

**Comment by fivesixzero** [comment]
This is weird, I'll look into it. Thanks for posting an issue!

What distro are you running? I've got a spare Pi Zero I can use to set up a test system.

**Comment by fivesixzero** [comment]
Ahh, I think I see what the problem is.

The `printf` command here is used to front-pad the binary output to a 20-character string. This is required since the hex-to-binary conversion via `dc` will output a raw binary value without any front-padding and the front-padding provides us with a consistent string to pick apart later. This is what happens with something like `0x0` or `0x50005`, which returns a 19-digit or shorter string.

But if our first bit is high, like it is with `0x80000`, then the output from `dc` is already 20 characters and `printf` flips out. 🤦

I just pushed a fix for this - [0b85bda](https://github.com/fivesixzero/telegraf-pi-bash/commit/0b85bdaa0bcbde9a7d1980364fd818450b36e684) - which adds a check on the length of the output from `dc` and, if it's already long enough, skips the front-padding. Tested this on a Pi running Bullseye and it works fine.

Let me know if this works!
